Website/app_code/code/Blog.cs

```diff
@@ -63,10 +63,10 @@ public static Post CurrentPost
         {
             if (HttpContext.Current.Items["currentpost"] == null && !string.IsNullOrEmpty(CurrentSlug))
             {
-                var post = Storage.GetAllPosts().FirstOrDefault(p => p.Slug == CurrentSlug);
+                var post = GetVisiblePosts().FirstOrDefault(p => p.Slug == CurrentSlug);
 
-                if (post != null && (post.IsPublished || HttpContext.Current.User.Identity.IsAuthenticated || (HttpContext.Current.Request.QueryString["key"] ?? string.Empty).Equals(post.ID, StringComparison.InvariantCultureIgnoreCase)))
```

Comment:
Hi Andrei,

The reason for the key parameter is to allow you to send a URL to an
unauthenticated person such that they can see an unpublished post.   Given
there is only a single authenticated user it allows you to get feedback on
an unpublished post (editorial or otherwise) without having to give away
the password.

(There's probably a job here to start documenting some of the features /
config options :-) )

Cheers
Sam

On Thu, Apr 23, 2015 at 4:58 PM, Andrei Dzimchuk <notifications@github.com>
wrote:

> In Website/app_code/code/Blog.cs
> <https://github.com/madskristensen/MiniBlog/pull/184#discussion_r28946688>
> :
>
> >
> > -                if (post != null && (post.IsPublished || HttpContext.Current.User.Identity.IsAuthenticated || (HttpContext.Current.Request.QueryString["key"] ?? string.Empty).Equals(post.ID, StringComparison.InvariantCultureIgnoreCase)))
>
> There was a conditional statement to include a post of its id is specified
> as a query string parameter called 'key' however I couldn't find a usage of
> that and it's not clear what scenario this was supposed to cover.
> Authenticated users will be able to see all posts and non-authenticated
> users will see posts that are published and their pubDates are in the past.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/madskristensen/MiniBlog/pull/184/files#r28946688>.
>